mvnw

```diff
@@ -199,6 +203,85 @@ if [ -z "$BASE_DIR" ]; then
   exit 1;
 fi
 
+##########################################################################################
+# Extension to allow automatically downloading the maven-wrapper.jar from Maven-central
```

Comment:
Since we bundle the jar as a part of this commit - do we still need to download them ? Or should we the jar that we have bundled ?

Reply:
No, we don't need to download a new jar. 
From the first glance this is a dead code. 

mvnw file is a dumb copy from
https://mvnrepository.com/artifact/org.apache.maven.wrapper/maven-wrapper-distribution/3.1.0

I've updated commit message.

Reply:
Can we point it to the source code location in github ?

Reply:
sure

Reply:
I've doublechecked, and seems this code is not that dead. I'll experiment with `wrapperUrl` property.

Reply:
Ah, ok, it's still takes in effect only if the jar is not present.

Reply:
> BTW how different is new maven-wrapper.jar from the one added in this PR - #9970
> What are the changes between these version?

If I understand correctly the major thing is the donation of the maven wrapper to the apache foundation. Other changes were minor, but they could be quite subtle ones and it's cheaper to keep wrapper up to date.
https://github.com/apache/maven-wrapper/compare/maven-wrapper-0.5.6...maven-wrapper-parent-3.1.0

Reply:
> Can we point it to the source code location in github ?

I've added a link to the gh repo to the commit message. 
`mvnw` file itself is generated during the build.